lib/openhab/core/items/persistence.rb

```diff
@@ -273,83 +360,293 @@ def timestamp
+        #   @return [void]
+        #   @since openHAB 4.2
+
+        # @!method remove_all_states_until(timestamp, service = nil)
+        #   Removes persisted data points from now until the given point in time.
+        #   @param [#to_zoned_date_time] timestamp The point in time until which to remove
+        #   @param [Symbol, String] service An optional persistence id instead of the default persistence service.
+        #   @return [void]
+        #   @since openHAB 4.2
+
+        # @!method remove_all_states_between(start, finish, service = nil)
+        #   Removes persisted data points between two points in time.
+        #   @param [#to_zoned_date_time] start The point in time from which to remove
+        #   @param [#to_zoned_date_time] finish The point in time to which to remove
+        #   @param [Symbol, String] service An optional persistence id instead of the default persistence service.
+        #   @return [void]
+        #   @since openHAB 4.2
+
+        #
+        # Persist item state to the persistence service
+        #
+        # @overload persist(service = nil)
+        #   Persists the current state of the item
+        #   @param [Symbol, String] service An optional persistence id instead of the default persistence service.
+        #   @return [void]
+        #
+        # @overload persist(timestamp, state, service = nil)
+        #   Persists a state at a given timestamp
+        #   @param [#to_zoned_date_time] timestamp The timestamp for the given state to be stored
+        #   @param [Types::State] state The state to be stored
+        #   @param [Symbol, String] service An optional persistence id instead of the default persistence service.
+        #   @return [void]
+        #   @since openHAB 4.2
+        #
+        # @overload persist(time_series, service = nil)
+        #   Persists a time series
+        #   @param [Types::TimeSeries] time_series The time series of states to be stored
+        #   @param [Symbol, String] service An optional persistence id instead of the default persistence service.
+        #   @return [void]
+        #   @since openHAB 4.2
+        #
+        def persist(*args)
+          # @deprecated OH 4.1 this if block content can be removed when dropping OH 4.1 support
+          if Core.version < Core::V4_2
+            raise ArgumentError, "wrong number of arguments (given #{args.size}, expected 0..1)" if args.size > 1
+
+            service = args.last || persistence_service
+            Actions::PersistenceExtensions.persist(self, service&.to_s)
+            return
+          end
+
+          first_arg = args.first
+          if first_arg.is_a?(TimeSeries)
+            if args.size > 2
+              raise ArgumentError,
+                    "wrong number of arguments to persist a time series (given #{args.size}, expected 1..2)"
+            end
+
+            service = args.last || persistence_service
```

Comment:
this still seems wrong. if `args` is _just_ the time_series, `args.last` will be the same as `args.first`, and you'll try to send that as the service. perhaps you mean `args[1]`?

Reply:
Yikes! I'm glad you caught this! I'm afk until tonight

Reply:
I've changed this to `args[1]`